src/System.IO.FileSystem/src/System/IO/FileStatus.Unix.cs

```diff
@@ -199,11 +201,20 @@ private void SetAccessWriteTimes(string path, long? accessTime, long? writeTime)
             // force a refresh so that we have an up-to-date times for values not being overwritten
             _fileStatusInitialized = -1;
             EnsureStatInitialized(path);
-            Interop.Sys.UTimBuf buf;
-            // we use utime() not utimensat() so we drop the subsecond part
-            buf.AcTime = accessTime ?? _fileStatus.ATime;
-            buf.ModTime = writeTime ?? _fileStatus.MTime;
-            Interop.CheckIo(Interop.Sys.UTime(path, ref buf), path, InitiallyDirectory);
+
+            // we use utimes()/utimensat() to set the accessTime and writeTime
+            Span<Interop.Sys.TimeSpec> buf = stackalloc Interop.Sys.TimeSpec[2];
+
+            // setting second part
+            buf[0].TvSec = accessTime ?? _fileStatus.ATime;
+            buf[1].TvSec = writeTime ?? _fileStatus.MTime;
+
+            // setting nanosecond part
+            buf[0].TvNsec = accessTime == null ? _fileStatus.ATimeNsec : 0;
+            buf[1].TvNsec = writeTime == null ? _fileStatus.MTimeNsec : 0;
```

Comment:
@Anipik I am attempting to backport this to Mono, so I am looking over this. I don't understand here how `.TvSec` and `.TvNSec` are both set to the same value (the `accessTime long` or the `writeTime long`).Before seeing this was merged for CoreFX 3.0,  I felt I had to change `SetAccessWriteTimes` to provide four values instead of two for subsecond resolution (please see my PR on the Mono CoreFX fork, which uses `timeval/utimes()` but should otherwise be similar: https://github.com/mono/corefx/pull/241/files)

Reply:
the key point here is that we will be either setting setting the lastwritetime or lastaccesstime which implies in the buf struct , one of the two elements will be zero and otherone will be set to the required value

Reply:
Ah, this looks to be corrected in https://github.com/dotnet/corefx/pull/31593

Reply:
Yeah but isn't this always using the subsecond part that's from `_fileStatus` (or 0)? So the users of this method (`SetLastAccessTime` and `SetLastWriteTime`) still can't pass-down the subsecond part of the `DateTimeOffset` they receive

Again, this appears fixed in the subsequent PR

Reply:
i just looked at the most recent version https://github.com/dotnet/corefx/blob/master/src/System.IO.FileSystem/src/System/IO/FileStatus.Unix.cs

Sry i deleted the previous comment.
So as i mentioned at one time we are setting only one of them(access or write), so we can just set the other one to the current value.


> So the users of this method (SetLastAccessTime and SetLastWriteTime) still can't pass-down the subsecond part of the DateTimeOffset they receive

The accuracy will be upto 100 nanoseconds, this is due to the limitations of DateTime infrastructure

Reply:
If there is an invocation `SetAccessWriteTimes(path, some_accesstime, null)` 
then:
`buf[0].TvSec = some_accessTime`
`buf[0].TvNSec = 0` 
`buf[1].TvSec = _fileStatus.MTime`
`buf[1].TvNSec = _fileStatus.MTimeNSec`

So this patch *preserves* the nanosecond part of the timestamp that's not being changed, but still can't properly set it for the timestamp that is being changed. 

The version on `master` appears to be correct, this is just a backporting concern.